conan/api/subapi/cache.py

```diff
@@ -16,6 +16,27 @@
 from conans.util.files import rmdir, gzopen_without_timestamps
 
 
+def _resolve_latest_ref(app, ref):
+    if ref.revision is None or ref.revision == "latest":
+        ref.revision = None
+        result = app.cache.get_latest_recipe_reference(ref)
+        if result is None:
+            raise ConanException(f"'{ref}' not found in cache")
+        ref = result
+    return ref
+
+
+def _resolve_latest_pref(app, pref):
+    pref.ref = _resolve_latest_ref(app, pref.ref)
+    if pref.revision is None or pref.revision == "latest":
+        pref.revision = None
+        result = app.cache.get_latest_package_reference(pref)
+        if result is None:
+            raise ConanException(f"'{pref.repr_notime()}' not found in cache")
+        pref = result
+    return pref
```

Comment:
Has something changed here, or is this only just a move for these functions? Why moving them? private implementations better at the end of the file not the top, specially for APIs